scripts/spack/packages/spheral/package.py

```diff
@@ -101,6 +102,25 @@ def _get_sys_type(self, spec):
             sys_type = env["SYS_TYPE"]
         return sys_type
 
+    def _get_arch(self):
+        host_platform = spack.platforms.host()
+        host_os = host_platform.operating_system("default_os")
+        host_target = host_platform.target("default_target")
+        architecture = spack.spec.ArchSpec((str(host_platform), str(host_os), str(host_target)))
+        spack_arch = str(architecture)
+        return spack_arch.strip()
+
+    # Create a name for the specific configuration being built
+    # This name is used to differentiate timings during performance testing
+    def _get_config_name(self, spec):
+        arch = self._get_arch()
+        config_name = f"{arch}_{spec.compiler.name}_{spec.compiler.version}"
+        if ("+mpi" in spec):
+            config_name += "_" + spec.format("{^mpi.name}_{^mpi.version}")
+        if ("+cuda" in spec):
+            config_name += "_" + spec.format("{^cuda.name}{^cuda.version}")
+        return config_name.replace(" ", "_")
+
```

Comment:
Some of this relies on Spack internal methods for determining things. We would have to extract those methods into CMake otherwise.

Reply:
I would prefer if we can do this during the CMake config stage at some point rather than relying on spack to generate a CMake string, but this is okay for now.